File: JavaFundamentals/src/Arrays/Exercise/TheLift.java
package Arrays.Exercise;

import java.util.Arrays;
import java.util.Scanner;
import java.util.stream.Collectors;

public class TheLift {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int peopleWaiting = Integer.parseInt(scanner.nextLine());
        int[] wagons = Arrays.stream(scanner.nextLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();

        for (int i = 0; i < wagons.length && peopleWaiting > 0; i++) {
            while (wagons[i] < 4 && peopleWaiting > 0) {
                wagons[i]++;
                peopleWaiting--;
            }
        }

        boolean hasEmptySpots = Arrays.stream(wagons).anyMatch(w -> w < 4);

        if (peopleWaiting == 0 && hasEmptySpots) {
            System.out.println("The lift has empty spots!");
            printWagons(wagons);
        } else if (peopleWaiting > 0) {
            System.out.printf("There isn't enough space! %d people in a queue!%n", peopleWaiting);
            printWagons(wagons);
        } else {
            printWagons(wagons);
        }
    }

    private static void printWagons(int[] wagons) {
        System.out.println(Arrays.stream(wagons)
                .mapToObj(String::valueOf)
                .collect(Collectors.joining(" ")));
    }
}

File: Fundamentals/Mid Exam Retake/the_lift.py
people_waiting = int(input())
all_seats_on_lift = list(map(int, input().split(' ')))
for wagon in range(len(all_seats_on_lift)):
    if all_seats_on_lift[wagon] <= 4:
        while all_seats_on_lift[wagon] != 4 and people_waiting > 0:
            all_seats_on_lift[wagon] += 1
            people_waiting -= 1
is_available_space = False
count_wagon = len(all_seats_on_lift)
for wagon in all_seats_on_lift:
    if wagon == 4:
        count_wagon -= 1
    if count_wagon == 0:
        is_available_space = True


if people_waiting == 0 and is_available_space:
    print(f'{" ".join([str(wagon) for wagon in all_seats_on_lift])}')
else:
    if people_waiting > 0:
        print(f"There isn't enough space! {people_waiting} people in a queue!")
        print(f"{' '.join([str(wagon) for wagon in all_seats_on_lift])}")
    else:
        print(f'The lift has empty spots!')
        print(f'{" ".join([str(wagon) for wagon in all_seats_on_lift])}')